src/DotNetWorker.Core/IBindingData.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.Functions.Worker.Core
+{
+    /// <summary>
+    /// A representation of a Microsoft.Azure.WebJobs.ParameterBindingData
+    /// </summary>
+    public interface IBindingData
```

Comment:
do we really need this interface? Doesn't seem like other types are following this pattern.

Reply:
We use this interface in the converters to determine if the converter can handle the request or not:

https://github.com/Azure/azure-functions-dotnet-worker/blob/92c5be67bcda0c3badd980ec2524f7c43eb7d859/extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs#L23-L26

Reply:
Hmm, that's fair. I think the intension was around project dependencies. ModelBindingData is defined in the Grpc project, and the extension projects will be [updated to reference the Core project](https://github.com/Azure/azure-functions-dotnet-worker/blob/568045c48fcda451a62b691f34c948ad3d429dbc/extensions/Worker.Extensions.Storage.Blobs/src/Worker.Extensions.Storage.Blobs.csproj#L19), this is so they have access to converter interfaces such as `IInputConverter`. And I believe we didn't want the extensions to take a dependency on the Grpc layer. If that isn't a good enough reason, then I am happy to modify this and work without an interface.

@fabiocav want to get your input here too as you helped make the initial interface decision

Reply:
Core types (`Worker.Core`) do not have a dependency on any types coming from the `Worker.Grpc` project, as that is one (of possibly many) transport implementations. Any time the gRPC implementation needs to pass its types to the infrastructure/core, it does so using the abstractions defined there. `IBindingData` is just that, the abstraction that is exposed in the public APIs in Core, which is what is exposed to converters and other components. Those components do not have a dependency on any other transport specific types.

Reply:
Guess I'm not clear on the layering here then. BindingData could itself probably just be in Worker.Core then? If you retain the interface, and we decide we need to add another bit of info later, is it easy for us to rev both the interface and the impl together?

Reply:
`BindingData` is an auto-generated type from the `proto` file (it's what is defined in the protobuf contract), internal and also implements some gRPC specific interfaces.

We could use an abstract class instead of the interface, as that would help mitigate the concern. That's what we've done for things like the `HttpRequestData`

Reply:
Done